kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/codec/ResponseDecoderTest.java

```diff
@@ -42,9 +53,67 @@ void testApiVersionsExactlyOneFrame_opaque(short apiVersion) throws Exception {
                 ApiKeys.API_VERSIONS::responseHeaderVersion,
                 v -> AbstractCodecTest.exampleResponseHeader(),
                 AbstractCodecTest::exampleApiVersionsResponse,
-                new KafkaResponseDecoder(mgr),
+                createResponseDecoder(mgr, DEFAULT_SOCKET_FRAME_MAX_SIZE_BYTES),
                 OpaqueResponseFrame.class, false),
                 "Unexpected correlation id");
     }
 
+    @NonNull
+    private static KafkaResponseDecoder createResponseDecoder(CorrelationManager mgr, int socketFrameMaxSizeBytes) {
+        return new KafkaResponseDecoder(mgr, socketFrameMaxSizeBytes);
+    }
+
+    @Test
+    void shouldThrowIfFirstIntGreaterThanMaxFrameSize() {
+        var mgr = new CorrelationManager(12);
+        int socketFrameMaxSizeBytes = 10;
```

Comment:
could we initialise these in a `@BeforeEach` to keep the test method focused on the case in hand?